ethdb/privateapi/ethbackend.go

```diff
@@ -270,6 +289,80 @@ func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.E
 	return nil, fmt.Errorf("unknown payload")
 }
 
+// EngineGetPayloadV1, retrieves previously assembled payload (Validators only)
+func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *remote.EngineForkChoiceUpdatedRequest) (*remote.EngineForkChoiceUpdatedReply, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("not a proof-of-stake chain")
+	}
+	if s.assembledBlock.Load() == nil {
+		return nil, fmt.Errorf("mining has not been enabled yet")
+	}
+	// Check if parent equate to the head
+	parent := gointerfaces.ConvertH256ToHash(req.Forkchoice.HeadBlockHash)
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	block := s.assembledBlock.Load().(*types.Block)
+
+	if parent != rawdb.ReadHeadBlockHash(tx) || block.Header().ParentHash != parent || atomic.LoadUint32(s.waitingForBeaconChain) == 0 {
+		return &remote.EngineForkChoiceUpdatedReply{
+			Status: string(Syncing),
+		}, nil
+	}
+	var baseFeeReply *types2.H256
+	if block.Header().BaseFee != nil {
+		var baseFee uint256.Int
+		baseFee.SetFromBig(block.Header().BaseFee)
+		baseFeeReply = gointerfaces.ConvertUint256IntToH256(&baseFee)
+	}
+	var encodedTransactions [][]byte
+	buf := bytes.NewBuffer(nil)
+
+	for _, tx := range block.Transactions() {
+		buf.Reset()
+
+		err := rlp.Encode(buf, tx)
+		if err != nil {
+			return nil, fmt.Errorf("broken tx rlp: %w", err)
+		}
+		encodedTransactions = append(encodedTransactions, common.CopyBytes(buf.Bytes()))
+	}
+	// Compute the correct block hash, by setting up the right header
+	header := block.Header()
+	header.Difficulty = serenity.SerenityDifficulty
+	header.Nonce = serenity.SerenityNonce
+	header.MixDigest = gointerfaces.ConvertH256ToHash(req.Prepare.Random)
+	header.Time = req.Prepare.Timestamp
+	header.Coinbase = gointerfaces.ConvertH160toAddress(req.Prepare.FeeRecipient)
```

Comment:
This is actually problematic. In the assembled block all transaction fees went to the original coinbase (not sure how that one is set). We cannot simply replace coinbase post `stage_mining_exec`. For now a quick hack would be to **not** set `header.Coinbase` to the FeeRecipient (and [the spec](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#payloadattributesv1) actually allows us to do it; there FeeRecipient is only a _suggested_ value).

Going forward, however, we need to reverse the order and start assembling the block with the right fee recipient after we have received `engine_forkchoiceUpdatedV1`, not beforehand.

Reply:
oh okay, that actually could be fixed if we run the mining stages directly from ethbackend, there is then also a bug now that i look at it when calling the timestamp and coinbase opcode they would be unmatching

Reply:
let me convert this to draft again, this needs to be fixed

Reply:
we also are not using correct gasLimit then so that might be an indication why verfication fails

Reply:
now it is fixed.